### PR TITLE
Add Hall of Fame page to beta version

### PR DESCRIPTION
**Pull request by michaelrambeau**
## Goal

Add missing _Hall of Fame_ page from the current version: https://bestofjs.org/hall-of-fame

I was not sure about where to put the link to the Hall of Fame: I reproduced what we do in production, adding a "More" button in the site header, to reveal more links.

Instead of this button, we could use the "Navigation Menu" component: https://ui.shadcn.com/docs/components/navigation-menu

I like the fact that you don't have to click to reveal the navigation items with this "Navigation Menu", it's triggered by a mouse `hover`, it helps to discover all the links of the application.

I added a loading state but I realized we never see it when the application is built: the page has no search parameters, so it's statically built and I think the loading state never shows up.

## How to test

Click on "More" button in the header and enjoy the company of great people 😄 

## Screenshots

image

image

### CODE REVIEW

File: apps/bestofjs-nextjs/src/app/hall-of-fame/loading.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Card } from "@/components/ui/card";
+import { Skeleton } from "@/components/ui/skeleton";
+import { PageHeading } from "@/components/core/typography";
+
+const memberNames = ["Evan", "Dan", "Sindre", "Ryan", "Kent", "TJ"];
```
Comment (zawarudo) [inline apps/bestofjs-nextjs/src/app/hall-of-fame/loading.tsx:5]: Committed to the repo, what an honor!

File: apps/bestofjs-nextjs/src/lib/global.d.ts [hunk 2]
```diff
@@ -104,13 +104,19 @@ declare namespace BestOfJS {
     projects: BestOfJS.Project[];
   }
 
-  type HallOfFameMember = {
-    username: string;
+  interface RawHallOfFameMember {
```
Comment (zawarudo) [inline apps/bestofjs-nextjs/src/lib/global.d.ts:107]: Raw means API's data model?

File: apps/bestofjs-nextjs/src/lib/global.d.ts [hunk 3]
```diff
@@ -104,13 +104,19 @@ declare namespace BestOfJS {
...
     bio: string;
     blog: string;
-  };
+    followers: numbers;
+    name: string;
+    projects: string[];
+    username: string;
+  }
+
+  interface HallOfFameMember extends RawHallOfFameMember {
```
Comment (zawarudo) [inline apps/bestofjs-nextjs/src/lib/global.d.ts:117]: and this is the App's data model?

**Review by zawarudo — APPROVED** [review]
Everything looks good, just some clarification questions.

File: apps/bestofjs-nextjs/src/lib/global.d.ts (hunk 2 above)
Comment (michaelrambeau) [reply apps/bestofjs-nextjs/src/lib/global.d.ts:107]: > 

The `Raw*` objects are the ones we get from the JSON files (the "static API" that is rebuilt everyday)

File: apps/bestofjs-nextjs/src/lib/global.d.ts (hunk 3 above)
Comment (michaelrambeau) [reply apps/bestofjs-nextjs/src/lib/global.d.ts:117]: And yes that is the populated object, the Hall of Game member with the related projects, ready to be consumed by the frontend application.

**Comment by michaelrambeau** [comment]
## Second iteration

- Added: project logo (for projects that have a SVG logo, the "Featured Projects")
- Removed: number of followers
- Link to GitHub profile set on the GitHub username, in orange

image

**Pull request merged**